File: src/main/java/com/mradziewicz/exchange/Controller/CurrencyController.java
package com.mradziewicz.exchange.Controller;

import com.mradziewicz.exchange.service.ExchangeRateService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class CurrencyController {

    private final ExchangeRateService exchangeRateService;

    public CurrencyController(ExchangeRateService exchangeRateService) {
        this.exchangeRateService = exchangeRateService;
    }

    //Sprzedajemy Obcą walutę
    @GetMapping("/buyCurrency/{curr}/{count}")
    public String getBuyExchangeRate(@PathVariable String curr, @PathVariable double count)  {
        return exchangeRateService.cantorExchangeCurrencyToPln(curr, count);
    }
    //Kupujemy obcą walutę
    @GetMapping("/sellCurrency/{curr}/{count}")
    public String getSellExchangeRate(@PathVariable String curr, @PathVariable double count){
        return exchangeRateService.cantorExchangePlnToCurrency(curr, count);
    }
    @GetMapping("/exchangeTwoCurrency/{curr1}/{curr2}/{count}")
    public String getSellExchangeTwoRate(@PathVariable String curr1, @PathVariable String curr2, @PathVariable double count){
        return exchangeRateService.cantorExchangeTwoCurrency(curr1, curr2, count);
    }
}

File: src/main/java/com/mradziewicz/exchange/Controller/CurrencyController.kt
package com.mradziewicz.exchange.Controller

import com.mradziewicz.exchange.exception.NotFoundException
import com.mradziewicz.exchange.service.ExchangeRateService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import javax.xml.stream.events.Characters

@RestController
class CurrencyController(var exchangeRateService: ExchangeRateService) {

    @GetMapping("/buyCurrency/{curr}/{amount}")
    fun getBuyExchangeRate(@PathVariable curr: String?, @PathVariable amount: Double): String {
            return exchangeRateService.cantorExchangeCurrencyToPln(curr!!, amount)

    }
    @GetMapping("/sellCurrency/{curr}/{amount}")
    fun getSellExchangeRate(@PathVariable curr: String?, @PathVariable amount: Double): String {
            return exchangeRateService.cantorExchangePlnToCurrency(curr!!, amount);

    }
    @GetMapping("/exchangeTwoCurrency/{sellCurrency}/{buyCyrrency}/{amount}")
    fun getSellExchangeTwoRate(@PathVariable sellCurrency: String, @PathVariable buyCyrrency: String, @PathVariable amount: Double): String{
        return exchangeRateService.cantorExchangeTwoCurrency(sellCurrency, buyCyrrency, amount);
    }
}